shared/src/main/scala/coop/rchain/metrics/Metrics.scala

```diff
@@ -8,6 +8,7 @@ import coop.rchain.metrics.Metrics.Source
 trait Span[F[_]] {
   def mark(name: String): F[Unit]
   def trace[A](source: Metrics.Source)(block: F[A]): F[A]
+  def trace[A](tag: String)(block: F[A])(implicit parentSource: Source): F[A]
```

Comment:
What is the difference between these definitions (source and parentSource)? Why this is not just extension method?

Reply:
I agree, this should be extension. But looks like common understanding that we should use macro, so this will be removed. Am i correct?

Reply:
Please create syntax, it should be also valid when we switch to macro.

Reply:
@tgrospic https://github.com/rchain/rchain/pull/3312/commits/bc95a9cc30333f465939f660c4f16de8415a1492